lib/schema_plus/indexes/active_record/connection_adapters/index_definition_5_2.rb

```diff
@@ -0,0 +1,52 @@
+module SchemaPlus::Indexes
+  module ActiveRecord
+    module ConnectionAdapters
+
+      module IndexDefinition
+        attr_accessor :orders
+
+        def initialize(*args) #:nodoc:
+          # same args as add_index(table_name, column_names, options)
+          if args.length == 3 and Hash === args.last
+            table_name, column_names, options = args + [{}]
+            super(
+              table_name, options[:name], options[:unique], column_names,
+              lengths: options[:length] || {},
+              orders: options[:orders] || {},
+              where: options[:where],
+              type: options[:type],
+              using: options[:using]
+            )
+          else # backwards compatibility
+            super
+          end
+          unless orders.blank?
+            # fill out orders with :asc when undefined.  make sure hash ordering
+            # follows column ordering.
+            if self.orders.is_a?(Hash)
+              self.orders = Hash[columns.map{|column| [column, orders[column] || :asc]}]
+            else
+              self.orders = Hash[columns.map{|column| [column, orders || :asc]}]
+            end
+          end
+        end
+
+        # tests if the corresponding indexes would be the same
+        def ==(other)
```

Comment:
Is this the same test as for earlier versions? If so I'd suggest placing this in a common module, and include the appropriate `initialize` based on the version of AR.

Reply:
It is..  And that makes sense.  @lowjoel What do you think about just one file and doing the Gem::Version check in that one file around the 2 versions of initialize?  Or I can do a class inheritance type thing.

Reply:
Option 1 is what I'd lean to, I don't think we'd need to resort to inheritance here.